src/R2_alg.jl

```diff
@@ -71,7 +71,8 @@ function R2(
   h::H,
   options::ROSolverOptions,
   x0::AbstractVector,
-) where {F <: Function, G <: Function, H}
+  T::R,
```

Comment:
If you type this, then `options` and `x` should also be parameterized by `R`.

Reply:
Also, let's give `T` a default value (say `-eps(R)^(1/3)`. Check that it is negative at the beginning of the function.